FinancialTransactionsApi/V1/Gateways/PostgreDbGateway.cs

```diff
@@ -25,9 +25,9 @@ public async Task<IEnumerable<Transaction>> GetByTargetId(string targetType, Gui
         {
             var spec = new GetTransactionByTargetTypeAndTargetIdSpecification(targetType, targetId, startDate, endDate);
 
-            var response = await _databaseContext.Transactions.Where(spec.Criteria).ToListAsync().ConfigureAwait(false);
+            var response = _databaseContext.Transactions.Where(spec.Criteria);
 
-            return response?.ToDomain();
+            return await Task.FromResult(response.AsEnumerable().ToDomain()).ConfigureAwait(false);
```

Comment:
Could the response be null here => `response?.AsEnumerable()`

Reply:
in fact, it is not possible the response object to be null as result because null cannot say that the sequence contains no elements, it will always say that the object reference is not set to an instance of an object. So, there are two possible results: the first to return exception with message: object reference not set to an instance of an object. Second one could be an empty sequence or sequence of objects.